src/server/v2.0/handler/gc.go

```diff
@@ -0,0 +1,198 @@
+package handler
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/goharbor/harbor/src/controller/gc"
+	"github.com/goharbor/harbor/src/core/config"
+	"github.com/goharbor/harbor/src/lib/errors"
+	"github.com/goharbor/harbor/src/pkg/scheduler"
+	"github.com/goharbor/harbor/src/pkg/task"
+	"github.com/goharbor/harbor/src/server/v2.0/handler/model"
+	"github.com/goharbor/harbor/src/server/v2.0/models"
+	operation "github.com/goharbor/harbor/src/server/v2.0/restapi/operations/gc"
+	"os"
+	"strings"
+)
+
+type gcAPI struct {
+	BaseAPI
+	gcCtr gc.Controller
+}
+
+func newGCAPI() *gcAPI {
+	return &gcAPI{
+		gcCtr: gc.NewController(),
+	}
+}
+
+func (g *gcAPI) CreateGCSchedule(ctx context.Context, params operation.CreateGCScheduleParams) middleware.Responder {
+	id, err := g.kick(ctx, params.Schedule.Schedule.Type, params.Schedule.Schedule.Cron, params.Schedule.Parameters)
+	if err != nil {
+		return g.SendError(ctx, err)
+	}
+	// replace the /api/v2.0/system/gc/schedule/{id} to /api/v2.0/system/gc/{id}
+	lastSlashIndex := strings.LastIndex(params.HTTPRequest.URL.Path, "/")
+	if lastSlashIndex != -1 {
+		location := fmt.Sprintf("%s/%d", params.HTTPRequest.URL.Path[:lastSlashIndex], id)
+		return operation.NewCreateGCScheduleCreated().WithLocation(location)
+	}
+	return operation.NewCreateGCScheduleCreated()
+}
+
+func (g *gcAPI) UpdateGCSchedule(ctx context.Context, params operation.UpdateGCScheduleParams) middleware.Responder {
+	_, err := g.kick(ctx, params.Schedule.Schedule.Type, params.Schedule.Schedule.Cron, params.Schedule.Parameters)
+	if err != nil {
+		return g.SendError(ctx, err)
+	}
+	return operation.NewUpdateGCScheduleOK()
+}
+
+func (g *gcAPI) kick(ctx context.Context, scheType string, cron string, parameters map[string]interface{}) (int64, error) {
+	// set the required parameters for GC
+	parameters["redis_url_reg"] = os.Getenv("_REDIS_URL_REG")
+	parameters["time_window"] = config.GetGCTimeWindow()
+
+	var err error
+	var id int64
+	switch scheType {
+	case ScheduleManual:
+		policy := gc.Policy{
+			ExtraAttrs: parameters,
+		}
+		if dryRun, ok := parameters["dry_run"].(bool); ok {
+			policy.DryRun = dryRun
+		}
+		if deleteUntagged, ok := parameters["delete_untagged"].(bool); ok {
+			policy.DeleteUntagged = deleteUntagged
+		}
+		id, err = g.gcCtr.Start(ctx, policy, task.ExecutionTriggerManual)
```

Comment:
check error

Reply:
it returned at line 86

Reply:
The error will be override by the code in line 73

Reply:
it's in the differernt code path, cannot be overrided.